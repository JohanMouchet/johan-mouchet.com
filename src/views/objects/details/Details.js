// @flow

import * as React from "react";
import cx from "classnames";
import "./Details.scss";

type Props = {
  summary?: Node,
  className?: string | Array<string> | Object,
  attributes?: { [key: string]: any },
  children: Node
};

const Details = ({
  summary = "Details",
  className,
  attributes,
  children
}: Props) => {
  const classNames = cx("o-details", className);

  return !children ? null : (
    <details className={classNames} {...attributes}>
      {summary && <summary className="o-details__summary">{summary}</summary>}
      <div className="o-details__panel">{children}</div>
    </details>
  );
};

export default Details;
