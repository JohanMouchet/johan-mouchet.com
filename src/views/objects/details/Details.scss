@import "assets/styles/settings/_settings";
@import "assets/styles/tools/_tools";

.o-details {
  $root: &;

  margin: t-vertical-rhythm(1) 0;
  border-bottom: 1px solid $s-color-gray;

  &[open] {
    #{$root}__summary {
      font-weight: 500;
      color: $s-color-link;
    }
  }

  /* Summary */
  &__summary {
    @include t-link($s-color-text-dimmed, "none", $hover-decoration: "none");

    display: block;
    padding: t-vertical-rhythm(0.5) 0;
    margin: 0;
    cursor: pointer;
    outline: 0;
  }

  /* Panel */
  &__panel {
    padding-bottom: t-vertical-rhythm(0.5);

    > *:first-child {
      margin-top: 0;
    }

    > *:last-child {
      margin-bottom: 0;
    }
  }

  &--compact {
    border-bottom: 0;

    #{$root}__summary {
      padding: 0;
    }
  }
}
