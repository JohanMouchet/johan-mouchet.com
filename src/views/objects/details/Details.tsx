import * as React from "react";
import cx from "classnames";
import "./Details.scss";

type Props = {
  summary?: string;
  variant?: "compact";
  className?: string | Array<string> | Object;
  attributes?: boolean | number | string;
  children: Node;
};

const Details = ({
  summary = "Details",
  variant,
  className,
  children,
  ...attributes
}: Props) => {
  const classNames = cx(
    "o-details",
    variant && `o-details--${variant}`,
    className
  );

  return !children ? null : (
    <details className={classNames} {...attributes}>
      {summary && <summary className="o-details__summary">{summary}</summary>}
      <div className="o-details__panel">{children}</div>
    </details>
  );
};

export default Details;
