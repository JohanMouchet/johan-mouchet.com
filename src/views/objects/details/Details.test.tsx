import React from "react";
import { render } from "@testing-library/react";
import Details from ".";

it("Renders <Details /> without crashing", () => {
  expect(
    render(
      <Details summary="Summary" variant="compact">
        children
      </Details>
    ).container.firstChild
  ).toMatchSnapshot();
});
