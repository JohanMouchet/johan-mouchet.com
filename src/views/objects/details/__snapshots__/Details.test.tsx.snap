// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Renders <Details /> without crashing 1`] = `
<details
  class="_o-details_8c2b27 _o-details--compact_8c2b27"
>
  <summary
    class="_o-details__summary_8c2b27"
  >
    <svg
      aria-hidden="true"
      class="_o-details__marker_8c2b27 _o-details__marker-open_8c2b27"
      viewBox="0 0 256 256"
    >
      <symbol
        fill="currentColor"
        id="689e5e3d"
        viewBox="0 0 256 256"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224,128a8,8,0,0,1-8,8H40a8,8,0,0,1,0-16H216A8,8,0,0,1,224,128Z"
        />
      </symbol>
      <use
        href="#689e5e3d"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="_o-details__marker_8c2b27 _o-details__marker-close_8c2b27"
      viewBox="0 0 256 256"
    >
      <symbol
        fill="currentColor"
        id="0222de37"
        viewBox="0 0 256 256"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224,128a8,8,0,0,1-8,8H136v80a8,8,0,0,1-16,0V136H40a8,8,0,0,1,0-16h80V40a8,8,0,0,1,16,0v80h80A8,8,0,0,1,224,128Z"
        />
      </symbol>
      <use
        href="#0222de37"
      />
    </svg>
    <span>
      Summary
    </span>
  </summary>
  <div
    class="_o-details__panel_8c2b27"
  >
    children
  </div>
</details>
`;
